pkg/processor/build/builder.go

```diff
@@ -657,6 +663,42 @@ func (b *Builder) createProcessorDockerfile() (string, error) {
 	return processorDockerfilePathInStaging, nil
 }
 
+// replace known keywords in docker command list with directives
+// runTime can be nil - used for injection testing
+func (b *Builder) preprocessBuildCommands(commands []string, runTime string) ([]string, error) {
+	var processedCommands []string
+
+	if runTime == "" {
+		runTime = time.Now().String()
+	}
+	knownKeywords := map[string]string{
+		"noCache": fmt.Sprintf("RUN echo %s > /dev/null", runTime),
+	}
+
+	commentPattern, err := b.getRuntimeCommentPattern(b.options.FunctionConfig.Spec.Runtime)
+	if err != nil {
+		return processedCommands, errors.Wrap(err, "Failed to get runtime comment pattern")
+	}
+
+	startCommandPattern := fmt.Sprintf("%s %s", commentPattern, inlineparser.StartBlockKeyword)
+
+	for _, line := range commands {
```

Comment:
`for _, thing := range things`

in this case: `for _, command := range commands`